File: src/2570. Merge Two 2D Arrays by Summing Values/soln.py
class Solution:
    def mergeArrays(self, nums1: List[List[int]], nums2: List[List[int]]) -> List[List[int]]:

        #ptrs for nums1 and nums2
        i,j=0,0
        result = []

        #if nums1[i] < nums2[j] then we add nums1[i] into the result else if theyre equal we add the values 

        while i < len(nums1) and j<len(nums2):
            if nums1[i][0] < nums2[j][0]:
                result.append(nums1[i])
                i+=1
            elif nums1[i][0] > nums2[j][0]:
                result.append(nums2[j])
                j+=1
            else:
                result.append([nums1[i][0], nums1[i][1]+nums2[j][1]])
                i+=1
                j+=1
        ##remaining values
        while i<len(nums1):
            result.append(nums1[i])
            i+=1
        while j<len(nums2):
            result.append(nums2[j])
            j+=1
        return result
        

File: src/2570. Merge Two 2D Arrays by Summing Values/soln.c
/**
 * Return an array of arrays of size *returnSize.
 * The sizes of the arrays are returned as *returnColumnSizes array.
 * Note: Both returned array and *columnSizes array must be malloced, assume caller calls free().
 */
int** mergeArrays(int** nums1, int nums1Size, int* nums1ColSize, int** nums2, int nums2Size, int* nums2ColSize, int* returnSize, int** returnColumnSizes) {

    int i=0,j=0;
    int maxSize = nums1Size + nums2Size;
    int** result = (int**)malloc(maxSize*sizeof(int*));

    *returnColumnSizes = (int*)malloc(maxSize*sizeof(int));

    *returnSize = 0;
    
    while (i < nums1Size && j < nums2Size) {
        if (nums1[i][0] < nums2[j][0]) {
            result[*returnSize] = (int*)malloc(2 * sizeof(int));
            result[*returnSize][0] = nums1[i][0];
            result[*returnSize][1] = nums1[i][1];
            (*returnColumnSizes)[*returnSize] = 2;
            (*returnSize)++;
            i++;
        } else if (nums1[i][0] > nums2[j][0]) {
            
            result[*returnSize] = (int*)malloc(2 * sizeof(int));
            result[*returnSize][0] = nums2[j][0];
            result[*returnSize][1] = nums2[j][1];
            (*returnColumnSizes)[*returnSize] = 2;
            (*returnSize)++;
            j++;
        } else {
            result[*returnSize] = (int*)malloc(2 * sizeof(int));
            result[*returnSize][0] = nums1[i][0];
            result[*returnSize][1] = nums1[i][1] + nums2[j][1];
            (*returnColumnSizes)[*returnSize] = 2;
            (*returnSize)++; 
            i++;
            j++;
        }
    }

    while (i < nums1Size) {
        result[*returnSize] = (int*)malloc(2 * sizeof(int));
        result[*returnSize][0] = nums1[i][0];
        result[*returnSize][1] = nums1[i][1];
        (*returnColumnSizes)[*returnSize] = 2;
        (*returnSize)++;
        i++;
    }

    
    while (j < nums2Size) {
        result[*returnSize] = (int*)malloc(2 * sizeof(int));
        result[*returnSize][0] = nums2[j][0];
        result[*returnSize][1] = nums2[j][1];
        (*returnColumnSizes)[*returnSize] = 2;
        (*returnSize)++;
        j++;
    }
        return result;  
}

